test/config/path-util.js

```diff
@@ -0,0 +1,155 @@
+const process = require('process');
+
+function createConfig() {
+    const runtimeConfig = new RuntimeConfig();
+    runtimeConfig.context = __dirname;
+    runtimeConfig.environment = 'dev';
+    runtimeConfig.babelRcFileExists = false;
+
+    return new WebpackConfig(runtimeConfig);
+}
+
+/**
+ * Some tests are very specific to different operating systems.
+ * We use this to only run them when needed.
+ *
+ * @returns {boolean}
+ */
+function isWindows() {
+    return process.platform === 'win32';
+}
+
+describe('path-util getContentBase()', () => {
+    describe('getContentBase()', () => {
+        it('contentBase is calculated correctly', function() {
+            if (isWindows()) {
+                this.skip();
+            }
+
+            const config = createConfig();
+            config.runtimeConfig.useDevServer = true;
+            config.runtimeConfig.devServerUrl = 'http://localhost:8080/';
+            config.outputPath = '/tmp/public/build';
+            config.setPublicPath('/build/');
+            config.addEntry('main', './main');
+
+            const actualContentBase = pathUtil.getContentBase(config);
+            // contentBase should point to the "document root", which
+            // is calculated as outputPath, but without the publicPath portion
+            expect(actualContentBase).to.equal('/tmp/public');
+        });
+
+        it('contentBase works ok with manifestKeyPrefix', function() {
+            if (isWindows()) {
+                this.skip();
+            }
+
+            const config = createConfig();
+            config.runtimeConfig.useDevServer = true;
+            config.runtimeConfig.devServerUrl = 'http://localhost:8080/';
+            config.outputPath = '/tmp/public/build';
+            config.setPublicPath('/subdirectory/build');
+            // this "fixes" the incompatibility between outputPath and publicPath
+            config.setManifestKeyPrefix('/build/');
+            config.addEntry('main', './main');
+
+            const actualContentBase = pathUtil.getContentBase(config);
+            expect(actualContentBase).to.equal('/tmp/public');
+        });
+
+        it('contentBase is calculated correctly on Windows', function() {
+            if (!isWindows()) {
+                this.skip();
+            }
+
+            const config = createConfig();
+            config.runtimeConfig.useDevServer = true;
+            config.runtimeConfig.devServerUrl = 'http://localhost:8080/';
+            config.outputPath = 'C:\\projects\\webpack-encore\\public\\build';
+            config.setPublicPath('/build/');
+            config.addEntry('main', './main');
+
+            const actualContentBase = pathUtil.getContentBase(config);
+            expect(actualContentBase).to.equal('C:\\projects\\webpack-encore\\public');
+        });
+    });
+
+    describe('generateManifestKeyPrefix', () => {
+        it('manifestKeyPrefix is correctly not required on windows', () => {
```

Comment:
Actually, based on your other suggestion - I've removed all of the skips and just add a bit of `isWindows` logic in each function to change the paths based on OS. This means all tests will run on both OS's - best of both worlds (and less duplication as you mentioned)